File: 2022/01/day01.go
package main

import (
    "bufio"
    "fmt"
    "log"
    "os"
    "sort"
    "strconv"
)


func partOne(filename string) {
}

func partTwo(filename string) {
}


func main() {
    if len(os.Args[1:]) < 1 {
        log.Fatal("Not enough arguments given!")
    }

    filename := os.Args[1]

    fmt.Println("Part One ####################")

    // Read file
    file, err := os.Open(filename)
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    // Init result
    var result int = 0

    calories := []int{0}
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        line := scanner.Text()

        if line == "" {
            calories = append(calories, 0)
            continue
        }

        lineInt, err := strconv.Atoi(line)
        if err != nil {
            log.Fatal(err)
        }
        
        calories[len(calories) - 1] += lineInt;
    }

    sort.Sort(sort.Reverse(sort.IntSlice(calories)))
    result = calories[0]
    fmt.Printf("Result: %d\n", result)

    fmt.Println("Part Two ####################")
    result = calories[0] + calories[1] + calories[2]
    fmt.Printf("Result: %d\n", result)
}

File: 2022/01/day01.py
#!/usr/bin/python3
import sys
from typing import List


def main(args):
    if len(args) < 2:
        print('[ERROR] Enter a file name!')
        print(f'\tUsage: python {args[0]} <filename>')
        sys.exit(1)

    filename = args[1]
    with open(filename, 'r') as f:
        lines = f.readlines()
        lines = [l.strip() for l in lines]

    print('Part One ' + '#'*41)
    calories = [0]

    for line in lines:
        if line != '':
            calories[-1] += int(line)
        else:
            calories.append(0)

    calories.sort(reverse=True)
    print(f'Max calories: {calories[0]}')

    print('Part Two ' + '#'*41)
    print(f'Max calories: {sum(calories[:3])}')


if __name__ == '__main__':
    main(sys.argv)
